### PR TITLE
restructure introducing framework

### PR DESCRIPTION
**Pull request by fgimenez**
Code reorg before changes for waiting for the guest cluster API up and adding more actual tests. It aligns with upstream framework concept https://github.com/kubernetes/kubernetes/tree/master/test/e2e/framework. Currently belongs to the same `integration` package as the tests (and because of that all the methods are private) but on further iterations it should be probably moved to its specific package.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -9,11 +10,6 @@ e2eTest: &e2eTest
 
     - run: ./e2e-harness setup --name=ci-awsop-${CIRCLE_SHA1:0:7}
 
-    - run:
-        name: additional setup required for compiling go on circleCI
-        command: |
-          mkdir -p /home/circleci/.go_workspace/src
-          cp -ar vendor/* /home/circleci/.go_workspace/src/
```
Comment (kopiczko) [inline .circleci/config.yml]: Why `working_directory` is needed and this is removed?

Comment (fgimenez) [reply .circleci/config.yml]: It is simpler to put the working directory inside `GOPATH` than copying the vendor directory there from another location, single directive and no copying involved.

File: integration/framework.go [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+		pod := pods.Items[0]
+		phase := pod.Status.Phase
+		if phase != v1.PodRunning {
+			return microerror.Maskf(unexpectedStatusPhaseError, "current status: %s", string(phase))
+		}
+		return nil
+	}
+}
+
+func (f *framework) activeNamespaceFunc(name string) func() error {
```
Comment (kopiczko) [inline integration/framework.go]: Those names are strange. Even though they are passed as arguments to other functions I think names should reflect what they do, not how they are used.

Comment (fgimenez) [reply integration/framework.go]: Makes total sense, thanks for pointing it out, fixed.

**Pull request merged**